packages/platform-android/src/commands/runAndroid/index.ts

```diff
@@ -324,15 +324,19 @@ function startServerInNewWindow(
     }
   }
   if (process.platform === 'linux') {
-    try {
-      return execa.sync(terminal, ['-e', `sh ${launchPackagerScript}`], {
-        ...procConfig,
-        detached: true,
-      });
-    } catch (error) {
-      // By default, the child shell process will be attached to the parent
-      return execa.sync('sh', [launchPackagerScript], procConfig);
-    }
+    // Awaiting this causes the CLI to hang indefinitely, so this must execute without .sync.
+    return execa(terminal, ['-e', 'sh', launchPackagerScript], {
```

Comment:
The CLI hangs indefinitely because launching packager is a persistent process, right? Maybe we could run the script in the void, unref it and then return?

Reply:
The problem of unref'ing the script is that it'll keep running once run-android quits, which will start a ghost packager in the background, not attached to any shell. Don't think that's what we want either